api/apps/api/src/modules/projects/projects.service.ts

```diff
@@ -167,14 +170,26 @@ export class ProjectsService {
     return await this.commandBus.execute(new ChangeBlmRange(projectId, range));
   }
 
-  async requestExport(projectId: string): Promise<string> {
-    // ACL slot
+  async requestExport(
+    projectId: string,
+    userId: string,
+  ): Promise<Either<typeof forbiddenError | typeof projectNotFound, string>> {
+    const userCanExportProject = await this.projectAclService.canExportProject(
+      userId,
+      projectId,
+    );
+    const isPublicProject = await this.projectChecker.isPublic(projectId);
+
+    if (isLeft(isPublicProject)) return left(projectNotFound);
 
-    return (
-      await this.commandBus.execute(
-        new ExportProject(new ResourceId(projectId)),
-      )
-    ).value;
+    if (!userCanExportProject && !isPublicProject.right) {
+      return left(forbiddenError);
+    }
```

Comment:
I am not too keen on this - here we're folding out logic that makes two different checks to know if we can honour the request to create an export for the project, and also returning an error in case the project is not found.

Even though this would break the nice pattern we have of using `this.projectAclService.canBlahBlah()`, I would try to wrap the checks we do (user has permissions on a private project *or* the project is public) in a function that abstracts away the compound check - all we need to know at this stage is whether we should let the request go ahead or not, leaving the exact logic to a function that knows what needs to be done in order to check this.

Reply:
@hotzevzl Mmm I have two main doubts with this approach. Firstly, are we sure we want the ProjectACLService to check more than roles? By definition, it seems correct, but I don't know if on a quick read of the code I would know that inside it is checking for roles and that the project is published. By naming, I'd usually expect that ProjectChecker actually checks if the project if public. But if what I mention about ProjectACLService checking more than roles is not really an issue, we can refactor it quickly without problems! 😄

Reply:
I'd lean towards keeping the ACL logic (whether this depends on roles or other conditions such as a project being published) in a single place, so I would favour wrapping the `isPublic()` check within `canExportProject()` (maybe we can name it `canExportProjectOrIsProjectPublic()` if we wanted to make things more explicit).

My main note about this bit here is actually about doing two checks (which could be fine) but also using the results of one of these checks to make sure the project actually exists.

Also right now we kind-of-duplicate this compound logic twice only, though as we add more features (cancelling a running export? deleting a still-valid export?, etc.) my feeling is that avoiding to unfold the logic to determine whether we can honour these requests in different service functions may help avoid accidental drifts on how we check these access control questions.

Reply:
Nice, we'll take this approach, ensure before all that the project exists and moving the rest to ACL service